src/utilities/do-try.test.ts

```diff
@@ -54,7 +76,7 @@ describe("do-try.ts", () => {
         it("When no errors occur, then catch handler is never called", async () => {
             // Arrange
             const catchHandler = jest.fn();
-            const workload = jest.fn();
+            const workload = async () => jest.fn()();

```

Comment:
I think you should be able to create a mock async function with `jest.fn().mockResolvedValue(something)`: https://jestjs.io/docs/en/mock-function-api#mockfnmockresolvedvaluevalue

Looks a bit nicer than this syntax 😛

Reply:
that's not accomplishing the same thing. The function passed into `try()` has to be an async function.

Reply:
Doesn't it?

```JS
test('async test', async () => {
  const asyncMock = jest
    .fn()
    .mockResolvedValue('default')
    .mockResolvedValueOnce('first call')
    .mockResolvedValueOnce('second call');

  await asyncMock(); // first call
  await asyncMock(); // second call
  await asyncMock(); // default
  await asyncMock(); // default
});
```

Reply:
@brandongregoryscott ah yeah, I could also just make it return a promise. duh. Not sure what I was thinking.